test/main/prepare_test.cpp

```diff
@@ -7,6 +7,48 @@ static void checkTuple(kuzu::processor::FlatTuple* tuple, const std::string& gro
     ASSERT_STREQ(tuple->toString().c_str(), groundTruth.c_str());
 }
 
+TEST_F(ApiTest, issueTest1) {
+    conn->query("CREATE NODE TABLE T(id SERIAL, name STRING, PRIMARY KEY(id));");
+    conn->query("CREATE (t:T {name: \"foo\"});");
+    auto preparedStatement = conn->prepare("MATCH (t:T {id: $p}) return t.name;");
+    auto result = conn->execute(preparedStatement.get(), std::make_pair(std::string("p"), 0));
+    ASSERT_TRUE(result->hasNext());
+    checkTuple(result->getNext().get(), "foo\n");
+    ASSERT_FALSE(result->hasNext());
+}
+
+TEST_F(ApiTest, issueTest2) {
+    conn->query("CREATE NODE TABLE NodeOne(id INT64, name STRING, PRIMARY KEY(id));");
+    conn->query("CREATE NODE TABLE NodeTwo(id INT64, name STRING, PRIMARY KEY(id));");
+    conn->query("CREATE Rel TABLE RelA(from NodeOne to NodeOne);");
+    conn->query("CREATE Rel TABLE RelB(from NodeTwo to NodeOne, name String);");
+    conn->query("CREATE (t: NodeOne {id:1, name: \"Alice\"});");
+    conn->query("CREATE (t: NodeOne {id:2, name: \"Jack\"});");
+    conn->query("CREATE (t: NodeTwo {id:3, name: \"Bob\"});");
+    auto preparedStatement = conn->prepare("MATCH (a:NodeOne { id: $a_id }),"
+                                           "(b:NodeTwo { id: $b_id }),"
+                                           "(c: NodeOne{ id: $c_id } )"
+                                           " MERGE"
+                                           " (a)-[:RelA]->(c),"
+                                           " (b)-[r:RelB { name: $my_param }]->(c)"
+                                           " return r.name;");
+    auto result = conn->execute(preparedStatement.get(), std::make_pair(std::string("a_id"), 1),
+        std::make_pair(std::string("b_id"), 3), std::make_pair(std::string("c_id"), 2),
+        std::make_pair(std::string("my_param"), "null"));
```

Comment:
hmmm did you check if DuckDB allow passing a `NULL` parameter?

Plus we are passing a string "null" here and everything is a string operation. If you are testing for string parameter use a different value.